### PR TITLE
added Spyder 5.x compatibility

### PR DESCRIPTION
**Pull request by skjerns**
I've added compatibility with Spyder 5 as discussed in #52 . This required some refactoring, however the core code should be mainly left untouched, except for some small bugs and API adaptations. I created a new sub-folder `spyder_line_profiler_5` which contains the code for Spyder 5. This way, both versions can still be installed. Currently the `setup.py` will simply look if Spyder 4 or Spyder 5 is installed and install the corresponding version.

If there is a more elegant solution let me know. We could just ditch Spyder 4 support, and/or let `pip` decide which version to install (however my experience with this approach is that it often leads to errors)

### CODE REVIEW

File: spyder_line_profiler_5/spyder/plugin.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    TABIFY = Plugins.Help
+    WIDGET_CLASS = SpyderLineProfiler5Widget
+    CONF_SECTION = NAME
+    CONF_WIDGET_CLASS = SpyderLineProfiler5ConfigPage
+
+    # --- Signals
+
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    def get_name(self):
```
Comment (ethanbb) [inline spyder_line_profiler_5/spyder/plugin.py]: I've just confirmed with a spyder developer that for compatibility with 5.2.0+, this has to be turned into a `@staticmethod`.

Comment (skjerns) [reply spyder_line_profiler_5/spyder/plugin.py]: Ok thanks, I will integrated that.

However, in the documentation this is not mentioned like that https://docs.spyder-ide.org/current/workshops/plugin-development.html#the-plugin-structure

Comment (skjerns) [reply spyder_line_profiler_5/spyder/plugin.py]: Before I change it: Is this the same for the other functions i.e. `get_description`, `get_icon` etc?

Comment (ethanbb) [reply spyder_line_profiler_5/spyder/plugin.py]: You would think, but according to this comment this is the only change to the API: https://github.com/spyder-ide/spyder/issues/16966#issuecomment-996079320

Comment (ccordoba12) [reply spyder_line_profiler_5/spyder/plugin.py]: > Before I change it: Is this the same for the other functions i.e. get_description, get_icon etc?

No, it's only necessary for `get_name`.

**Pull request closed without merging**